### PR TITLE
Implement database tests for subject table

### LINKED ISSUE #81: Add database integration tests for subjects
In the process of investigating database migration tools, we added a migration schema for `subjects`. A subject is the target of a compliance scan (a linux server, laptop, or infrastructure).

We need to add tests to ensure this schema does what we expect now that we have a test harness of integrated database tests.

https://github.com/rhmdnd/compserv/commit/3b2d322b536975cbbfea7de3b966b87fd36242d8

### PR DESCRIPTION
**Pull request by rhmdnd**
Fixes #81

### CODE REVIEW

File: migrations/000001_create_subject_table.up.sql [hunk 1]
```diff
@@ -1,5 +1,5 @@
 CREATE TABLE IF NOT EXISTS subjects (
```
Comment (JAORMX) [inline migrations/000001_create_subject_table.up.sql:1]: Let's add documentation about what a Subject can be. I recall you mentioned that a subject may be the entity that's being audited for compliance. e.g. a server, a cluster, and so on. Let's mention this in documentation.

File: migrations/000001_create_subject_table.up.sql [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
-   id serial PRIMARY KEY,
+   id uuid PRIMARY KEY,
    name VARCHAR(255),
    type VARCHAR(50)
 );
```
Comment (JAORMX) [inline migrations/000001_create_subject_table.up.sql:5]: What about adding metadata to better help locate a subject? While we use very consistent naming for our servers (just from the name we'll know a lot about what server and where it is) some other places treat their servers more like cattle and have very vague names, which can get confusing. I'd suggest adding a json blob column that may store metadata for subjects.

File: tests/integration/integration_test.go [hunk 3]
```diff
@@ -66,18 +79,114 @@ func getMigrationHelper(t *testing.T) *migrate.Migrate {
...
+	assert.NotEmpty(t, err, "Shouldn't be able to insert type values longer than 50 characters")
+	// Drop the database instead of downgrading since we don't need the
+	// data anyway
+	if err := m.Drop(); err != nil {
+		t.Fatalf("Unable to drop database: %s", err)
+	}
 }
 
-func TestMigration(t *testing.T) { // nolint:paralleltest // database tests should not run in parallel
+func TestMigration(t *testing.T) { // nolint:paralleltest // database tests should run serially
```
Comment (JAORMX) [inline tests/integration/integration_test.go:189]: why do they need to run serially?

File: tests/integration/integration_test.go [hunk 4]
```diff
@@ -66,18 +79,114 @@ func getMigrationHelper(t *testing.T) *migrate.Migrate {
...
 // Each test assumes the database is unmanaged. The test is responsible for
 // setting up the state it requires for its test logic. This keeps the
 // getMigrationHelper() method clean of any assumptions about what the tests
 // expect from it.
-func TestSubject(t *testing.T) { // nolint:paralleltest // database tests should not run in parallel
-	getMigrationHelper(t)
-	// m.Up()
-	// Test logic
-	// m.Down()
+func TestInsertSubjectSucceeds(t *testing.T) { // nolint:paralleltest // database tests should run serially
```
Comment (JAORMX) [inline tests/integration/integration_test.go:97]: why do tests need to run serially?

**Review by JAORMX — COMMENTED** [review]
let's give this some thought... subjects are the entity that's being evaluated, right? If so... how will we divide OpenShift clusters, which have different attributes per machine pool? Will they have different subjects for each pool? Then, how will we add a relationship? e.g. worker nodes and control plane nodes belong to the same cluster.

File: tests/integration/integration_test.go (hunk 4 above)
Comment (rhmdnd) [reply tests/integration/integration_test.go:97]: I'll add a comment. But each test is responsible for upgrading and downgrading the database to the right version for testing. If we run these tests concurrently against the same database, we will get issues with the database state (one test migrating to a database version that is too old for another test).

File: tests/integration/integration_test.go (hunk 3 above)
Comment (rhmdnd) [reply tests/integration/integration_test.go:189]: I'll add a comment for this so it's more clear.

File: migrations/000001_create_subject_table.up.sql (hunk 1 above)
Comment (rhmdnd) [reply migrations/000001_create_subject_table.up.sql:1]: FWIW - I do have an issue open to discuss the overall schema if we want to iron some of that out there (and take propose the outcome as documentation).

https://github.com/rhmdnd/compserv/issues/84

**Review by Vincent056 — APPROVED** [review]
/lgtm

**Comment by rhmdnd** [comment]
Following up that this will likely need more work and test coverage depending on the outcomes of https://github.com/rhmdnd/compserv/issues/84

At the very least - this should serve as an example for other contributors to build more testing on as we develop more schemas.

**Pull request merged**